### PR TITLE
Add client.transcripts.list_by_url & update README.md

### PR DESCRIPTION
**Pull request by Swimburger**
Add client.transcripts.list_by_url & update README.md

### CODE REVIEW

**Comment by Swimburger** [comment]
@armandobelardo Does `list_by_url` look good? Is this naming good by Ruby conventions?

**Comment by armandobelardo** [comment]
Thank you for adding this!
You'll also want to add an import to the root client: https://github.com/AssemblyAI/assemblyai-ruby-sdk/blob/main/lib/assemblyai.rb (see line 8, apologies I couldn't link to the line on mobile)

After both changes I can test this out locally

File: lib/assemblyai/transcripts/list_by_url_client.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+require_relative "types/transcript_list"
+require "async"
+
+module AssemblyAI
+  # Retrieve a list of transcripts you created
+  #
+  # @param url [String] The URL to retrieve the transcript list from
+  # @param request_options [RequestOptions]
+  # @return [Transcripts::TranscriptList]
+  def list_by_url(url: nil, request_options: nil)
```
[the remark this answers is not in the document]
Comment (Swimburger) [reply lib/assemblyai/transcripts/list_by_url_client.rb]: Good catch, I accidentally removed that.

**Comment by Swimburger** [comment]
Does it work now?

**Comment by Swimburger** [comment]
Thanks for the updates and test!

**Pull request merged**